queue-rabbitmq/src/main/java/com/cisco/oss/foundation/message/RabbitMQMessagingFactory.java

```diff
@@ -267,7 +275,57 @@ public void onRecoveryFailure(Consumer consumer, Channel channel, Throwable fail
             ConnectionOptions options = new ConnectionOptions()
                     .withAddresses(addresses.toArray(addrs));
 
-            Configuration configuration = ConfigurationFactory.getConfiguration();
+            final boolean metricsAndMonitoringIsEnabled = configuration.getBoolean("service.rabbitmq.MetricsAndMonitoringJmx.isEnabled", false);
+
+            if (metricsAndMonitoringIsEnabled){
+                MetricRegistry registry = new MetricRegistry();
+                final ConnectionFactory connectionFactory = options.getConnectionFactory();
+                StandardMetricsCollector metrics = new StandardMetricsCollector(registry);
+                connectionFactory.setMetricsCollector(metrics);
+
+                JmxReporter reporter = JmxReporter
+                        .forRegistry(registry)
+                        .inDomain("com.rabbitmq.client.jmx")
+                        .build();
+                reporter.start();
+            }
+
+            final boolean useNio = configuration.getBoolean("service.rabbitmq.useNio", false);
+
+            if (useNio) {
```

Comment:
does the rest of the flow stay the same when Nio is enabled or does the connect api need to be used differently? in other words do we need an else for this 'if'?